static/docs/commands-reference/push.md

```diff
@@ -1,42 +1,115 @@
 # push
 
-This command pushes all data file caches related to the current Git branch to
-the remote storage.
+Uploads files and directories from the current branch in the local workspace to
+the [remote storage]('doc/commands-reference/remote').
```

Comment:
We push data from cache based on DVC files in the working space. For example, (let's double check this), if I run something with --no-commit and then `dvc push`, data from the working space won't be uploaded to remote. Again, let's confirm and let's come with a better summary.